lib/ansible/utils/encrypt.py

```diff
@@ -78,9 +78,9 @@ class BaseHash(object):
     algo = namedtuple('algo', ['crypt_id', 'salt_size', 'implicit_rounds', 'salt_exact', 'implicit_ident'])
     algorithms = {
         'md5_crypt': algo(crypt_id='1', salt_size=8, implicit_rounds=None, salt_exact=False, implicit_ident=None),
-        'bcrypt': algo(crypt_id='2a', salt_size=22, implicit_rounds=None, salt_exact=True, implicit_ident='2a'),
-        'sha256_crypt': algo(crypt_id='5', salt_size=16, implicit_rounds=5000, salt_exact=False, implicit_ident=None),
-        'sha512_crypt': algo(crypt_id='6', salt_size=16, implicit_rounds=5000, salt_exact=False, implicit_ident=None),
+        'bcrypt': algo(crypt_id='2b', salt_size=22, implicit_rounds=12, salt_exact=True, implicit_ident='2b'),
```

Comment:
`2a` is a legacy value. Changing it to `2b` will be a breaking change.

cc @bcoca @sivel @samdoran What do you think?

Reply:
https://passlib.readthedocs.io/en/stable/lib/passlib.hash.bcrypt.html:
`"2a" - some implementations suffered from rare security flaws, replaced by 2b.`

Reply:
https://passlib.readthedocs.io/en/stable/history/1.7.html#id15, `1.7.0 (2016-11-22)`:
`bcrypt and bcrypt_sha256 now default to the “2b” format.`

Reply:
@konstruktoid Yes, I agree that `2a` suffers security flaws. But in the context of Ansible, we need to maintain backward compatibility. I think we can come up with [porting guide](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_core_2.12.rst#id9) so that people can migrate from `2a` to `2b`. Let us wait for others to comment.

Reply:
I would be in favor of changing this to `2b` and adding a porting guide entry.

Reply:
What would that entry include for example, since changing `2a` to `2b` only would affect newly created hashes?
"Old" `password_hash` will still work.

Reply:
Just state that the default when unspecified has changed. If anyone needs 2a specifically, they must specify it with `password_hash`.